cocotb/share/lib/gpi/GpiCommon.cpp

```diff
@@ -116,16 +129,22 @@ void gpi_embed_init(gpi_sim_info_t *info)
 }
 
 void gpi_embed_end()
-
 {
     embed_sim_event(SIM_FAIL, "Simulator shutdown prematurely");
+    gpi_cleanup();
 }
 
 void gpi_sim_end()
 {
     registered_impls[0]->sim_end();
 }
 
+void gpi_cleanup(void)
+{
+    CLEAR_STORE();
```

Comment:
What's the point in this?

Reply:
If SINGLETON_HANDLES is defined, we store sim handles and re-use them.
This will delete those handle objects, and in the case of VHPI, release the handles back to the simulator (see VhpiCbHdl.cpp changes).

Reply:
Not paying attention, sorry - feel free to resolve this comment chain.